src/app/components/client/csat_survey/CsatSurvey.tsx

```diff
@@ -79,14 +82,29 @@ export const CsatSurvey = (props: CsatSurveyProps) => {
     }
   });
 
+  const isPetitionCsatBanner =
+    currentSurvey.localDismissalId.includes("petition_banner");
+  const hasDismissedDataPrivacyPetitionBanner =
+    typeof Object.keys(cookies.getAll()).find((cookieName) =>
+      cookieName.includes("data_privacy_petition_banner"),
+    ) !== "undefined";
```

Comment:
Other question: should the local dismissal ID maybe include the user's subscriber ID? So that if multiple users use Monitor on the same device, one user dismissing it won't dismiss it for the other user? Edge case, but still.

Reply:
Using a custom `useLocalDismissal` hook here makes sense. Good point about including the subscriber ID to not run into the edge case described by you also. I’ve refactored this in [8c7888f](https://github.com/mozilla/blurts-server/pull/4775/commits/8c7888f80eb5e8514990e7b266dec8f9ac2a42fc) and [8d7f3b4](https://github.com/mozilla/blurts-server/pull/4775/commits/8d7f3b44e2b51191ad9f90e3eb778d0ff9f83b99).